### DISCUSSION TITLE
Adding clan chat to discord channels by using DiscordSRV

### DISCUSSION THREAD

**Discussion by Tomut0** [RoinujNosde/SimpleClans#196] [Ideas]
It will be cool, if **verified** clans will have an own chat in discord channel.
I think, you should use [DiscordSRV](https://www.spigotmc.org/resources/discordsrv.18494/) for that.

**Comment by RoinujNosde** [comment]
- There should be a LinkAccountEvent (didn't check the API), when that happens the player could be added to its clan's channel.
- Clan disband or clan create should create/remove channels.
- Joining or leaving a clan should update the participants as well.
- and, of course, the chat event for sending the messages.

**Comment by RoinujNosde** [comment]
Maybe there could be channels for ally chat too?

- the name would be a string joining the participant clans' tags.

**Reply by Tomut0** [reply]
> the name would be a string joining the participant clans' tags.

I think, this will looks like:
```
category name (the tags of the participant clans)
- voice-channel
- chat-channel
```
**Why category?**
As you know, channels, which has category inherits the rules from its.
So, it makes easier to create a new channel inside of category.

Example:
```
Roar,ZiP,Lead,Best...
- Voice-channel
- Chat-channel
```

The problem might be that DiscordSRV couldn't has an option to create a category and manupulate with rules. (didn't check it)

**Reply by RoinujNosde** [reply]
Yeah, categories are better for this.

**Comment by Tomut0** [comment]
Should permanent clans has individual marker? (for example, `<clan name> | P`, - where P is Permanent)

**Reply by RoinujNosde** [reply]
In the plugin itself (commands, chat, etc) there is no such suffix, so I think, no.
They could be ordered first in the channels list.

**Comment by Tomut0** [comment]
The mind, which I can't reach: 
> if they want to share the db, the category names are the same
@RoinujNosde 

(Cross-server SimpleClans)

**Comment by Tomut0** [comment]
DiscordSRV support was added in: #215

**Comment by Tomut0** [comment]
Clan' bulletin boards as embed discord message
